uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/manage-leave-requests.component.js

```diff
@@ -436,6 +436,32 @@ define([
     }
 
     /**
+     * Applies the filters when pending request is checked
+     *
+     * @param {Array} statusFilter
+     * @return {Array}
+     */
+    function preparePendingRequestOnlyFilters (statusFilter) {
+      var pendingRequestFilters = [
+        getStatusValueFromName(sharedSettings.statusNames.moreInformationRequired),
+        getStatusValueFromName(sharedSettings.statusNames.awaitingApproval)
+      ];
+
+      // Remove others filters
+      statusFilter = statusFilter.filter(function (status) {
```

Comment:
It seems that you are removing any status filter that it's not either "more info required" or "awaiting approval", then why can't you simply do this

```js
function prepareStatusFilter (filterByStatus) {
  // ...
  if (filters.pending_requests) {
    // add only "more info required" and "awaiting approval" status"
  } else if (filterByStatus && filters.leaveStatus && filters.leaveStatus.value) {
    // etc.
  }
}
```
instead of having to remove the previously-added filters in `preparePendingRequestOnlyFilters` ?